### PR TITLE
Support forwarding performance tests

### PR DESCRIPTION
**Pull request by jumaffre**
This PR adds support for running multi-node performance tests with forwarding. It mostly is a bunch of minor changes to the remote client python infra (e.g. copying all the _recv and _sent files back to the client) and client (`--fast-start` so that clients don't wait for a stable commit before starting - otherwise, clients will run sequentially). 

I've also re-enabled pushing performance data (tx-times and metrics) to Blob Storage.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -37,12 +37,6 @@ commands:
 
   gather_and_push_coverage:
     steps:
-      - run:
```
Comment (jumaffre) [inline .circleci/config.yml]: This is not necessary anymore for pushing coverage to codecov.

File: samples/apps/smallbank/smallbank.cmake [hunk 2]
```diff
@@ -21,6 +21,7 @@ if(BUILD_TESTS)
     CLIENT_BIN ./small_bank_client
     VERIFICATION_FILE ${CMAKE_CURRENT_LIST_DIR}/tests/verify_small_bank.json
     ADDITIONAL_ARGS
+      --label Small_Bank_ClientCpp
```
Comment (jumaffre) [inline samples/apps/smallbank/smallbank.cmake]: We currently have this small bank test recorded under a different name on Blob Storage so I am reverting its label back to what it used to be.

File: samples/apps/smallbank/smallbank.cmake [hunk 3]
```diff
@@ -47,4 +48,21 @@ if(BUILD_TESTS)
       --max-writes-ahead 1 --warmup 1000 --cooldown 1000
       --metrics-file small_bank_wc_metrics.json
   )
+
+  # It is better to run performance tests with forwarding on different machines
```
Comment (achamayou) [inline samples/apps/smallbank/smallbank.cmake]: Do we still need the warmup/cooldown then?

Comment (achamayou) [reply samples/apps/smallbank/smallbank.cmake]: Let's take it out in a separate PR as discussed.

Comment (jumaffre) [reply samples/apps/smallbank/smallbank.cmake]: I'll address this in another PR.

File: tests/infra/rates.py [hunk 4]
```diff
@@ -7,7 +7,7 @@
 
 from loguru import logger as LOG
 
-COMMIT_COUNT_CUTTOF = 10
+COMMIT_COUNT_CUTOFF = 10
```
Comment (eddyashton) [inline tests/infra/rates.py]: @eddyashton likes this

**Pull request merged**